src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/Helpers/StringSearchValuesHelper.cs

```diff
@@ -0,0 +1,294 @@
+                }
+
+                ref byte first = ref Unsafe.As<char, byte>(ref matchStart);
+                ref byte second = ref Unsafe.As<char, byte>(ref candidate.GetRawStringData());
+                nuint byteLength = (nuint)(uint)candidate.Length * 2;
+
+                if (TValueLength.AtLeast4Chars)
+                {
+                    nuint offset = byteLength - sizeof(ulong);
+                    ulong differentBits = (Unsafe.ReadUnaligned<ulong>(ref first) & ~0x20002000200020u) - Unsafe.ReadUnaligned<ulong>(ref second);
+                    differentBits |= (Unsafe.ReadUnaligned<ulong>(ref Unsafe.Add(ref first, offset)) & ~0x20002000200020u) - Unsafe.ReadUnaligned<ulong>(ref Unsafe.Add(ref second, offset));
+                    return differentBits == 0;
+                }
+                else
+                {
+                    nuint offset = byteLength - sizeof(uint);
+                    uint differentBits = (Unsafe.ReadUnaligned<uint>(ref first) & ~0x200020u) - Unsafe.ReadUnaligned<uint>(ref second);
+                    differentBits |= (Unsafe.ReadUnaligned<uint>(ref Unsafe.Add(ref first, offset)) & ~0x200020u) - Unsafe.ReadUnaligned<uint>(ref Unsafe.Add(ref second, offset));
+                    return differentBits == 0;
+                }
+            }
+        }
+
+        public readonly struct CaseInsensitiveAscii : ICaseSensitivity
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static char TransformInput(char input) => TextInfo.ToUpperAsciiInvariant(input);
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector128<byte> TransformInput(Vector128<byte> input)
+            {
+                Vector128<byte> subtraction = Vector128.Create((byte)(128 + 'a'));
+                Vector128<byte> comparison = Vector128.Create((byte)(128 + 26));
+                Vector128<byte> caseConversion = Vector128.Create((byte)0x20);
+
+                Vector128<byte> matches = Vector128.LessThan((input - subtraction).AsSByte(), comparison.AsSByte()).AsByte();
+                return input ^ (matches & caseConversion);
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector256<byte> TransformInput(Vector256<byte> input)
+            {
+                Vector256<byte> subtraction = Vector256.Create((byte)(128 + 'a'));
+                Vector256<byte> comparison = Vector256.Create((byte)(128 + 26));
+                Vector256<byte> caseConversion = Vector256.Create((byte)0x20);
+
+                Vector256<byte> matches = Vector256.LessThan((input - subtraction).AsSByte(), comparison.AsSByte()).AsByte();
+                return input ^ (matches & caseConversion);
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector512<byte> TransformInput(Vector512<byte> input)
```

Comment:
@tannergooding, this PR feels like a really good candidate for exploring ways of reducing duplication between Vector128/256/512.